src/Controller/FrontendController.php

```diff
@@ -70,16 +70,18 @@ public function shareAction(): RedirectResponse
      *
      * @Route("/_contao/login", name="contao_frontend_login")
      */
-    public function loginAction(): void
+    public function loginAction(): RedirectResponse
     {
+        return $this->redirectToRoute('contao_root');
```

Comment:
This will potentially redirect the user to a different language than the one they were just browsing. If I e.g. log in on contao.org, switch to English and then log out, I will be redirected to the German start page.

Reply:
This response would only be triggered if someone manually opens the `_contao/login` without a POST request. In that case the firewall listener will not be triggered but the controller is. Without this, one would get an exception that the controller did not return a response. I don't think we need to handle correct locale redirection for URL manipulation?

Reply:
Nope.